glusterfsd/src/glusterfsd-mgmt.c

```diff
@@ -355,41 +356,20 @@ glusterfs_handle_terminate(rpcsvc_request_t *req)
     }
 
     glusterfs_terminate_response_send(req, 0);
-    for (trav_p = &top->children; *trav_p; trav_p = &(*trav_p)->next) {
-        tvictim = (*trav_p)->xlator;
-        if (!tvictim->cleanup_starting &&
-            !strcmp(tvictim->name, xlator_req.name)) {
-            continue;
-        }
-        if (!tvictim->cleanup_starting) {
-            still_bricks_attached = _gf_true;
-            break;
-        }
-    }
-    if (!still_bricks_attached) {
-        gf_log(THIS->name, GF_LOG_INFO,
-               "terminating after loss of last child %s", xlator_req.name);
-        rpc_clnt_mgmt_pmap_signout(glusterfsd_ctx, xlator_req.name);
-        kill(getpid(), SIGTERM);
-    } else {
-        /* TODO cleanup sequence needs to be done properly for
-           Quota and Changelog
-        */
-        if (victim->cleanup_starting)
-            goto err;
-
-        rpc_clnt_mgmt_pmap_signout(glusterfsd_ctx, xlator_req.name);
-        victim->cleanup_starting = 1;
+    if (victim->cleanup_starting)
+        goto err;
```

Comment:
I think this cannot happen. if `cleanup_starting` is set, `victim` should be NULL, so we would have entered the `if` at line 346 and we'll never reach here. If we want to detect this condition, we need to do it in line 336.

Reply:
we are already doing it
if (!tmp->cleanup_starting &&
                    strcmp(tmp->name, xlator_req.name) == 0) {
                    victim = tmp;
                }

Reply:
If the volume we are trying to stop has already started cleanup, `victim->cleanup_starting` will be true, but in this case, we would have never reached line 339, so `victim` will be NULL and we'll never reach this check.